src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -889,6 +889,9 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
                     if ((object)associatedField != null)
                     {
                         Debug.Assert((object)associatedField.AssociatedSymbol != null);
+
+                        associatedField = associatedField.TupleUnderlyingField ?? associatedField;
```

Comment:
This pattern pops up a bit. Is it worth adding a new member like `TulpeUnderlyingFieldOrSelf` to encourage this as a best practice going forward?

Reply:
> This pattern pops up a bit. Is it worth adding a new member like TulpeUnderlyingFieldOrSelf to encourage this as a best practice going forward?

Are you suggesting to add one? I didn't get an urge to add one.

Reply:
Reading the PR I was wondering to myself:

> How will future developers know this is an important problem? If we had `TupleUnderlyingFieldOrSelf` maybe it would be a clue that this is important. 

I don't feel strongly about this. It's just a thought I had reading through this PR.

Reply:
> How will future developers know this is an important problem?

I guess I am not sure what is the problem to be aware of. There is an API TupleUnderlyingField, it is docummented to return null in some scenarios. What is there to miss or be confused about?

Reply:
Mostly that in the case the return is `null` then using `this` is the generally the correct fallback. That can also be inferred by looking through these cases. 

As I said though, this isn't a big concern. Mostly just thinking out loud as I'm reading the PR.